File: Daily Challenge/January 2025/Number of Ways to Split Array.cs
public class Solution {
  public int WaysToSplitArray(int[] nums) {
    int n = nums.Length;
    long[] prefixSum = new long[n];

    for (int i = 0; i < n; ++i) {
      prefixSum[i] = nums[i] + (i > 0 ? prefixSum[i - 1] : 0);
    }

    int splits = 0;
    for (int i = 0; i < (n - 1); ++i) {
      if (prefixSum[i] >= prefixSum[n - 1] - prefixSum[i]) splits++;
    }

    return splits;
  }
}

File: Daily Challenge/January 2025/Number of Ways to Split Array.cpp
class Solution {
public:
  int waysToSplitArray(vector<int>& nums) {
    int n = nums.size();
    vector<long long> prefix(n, 0);
    for (int i = 0; i < n; ++i) {
      prefix[i] = nums[i];
      if (i > 0) {
        prefix[i] += prefix[i - 1];
      }
    }

    int splits = 0;
    for (int i = 0; i < (n - 1); ++i) {
      if (prefix[i] >= (prefix[n - 1] - prefix[i])) splits++;
    }

    return splits;
  }
};